src/datumaro/components/annotation.py

```diff
@@ -204,7 +204,8 @@ def add(
         attributes: Optional[Set[str]] = None,
     ) -> int:
         assert name
-        assert name not in self._indices, name
+        if name in self._indices:
```

Comment:
It seems to raise an error when an user is trying to add the duplicated label `name`. It will introduce a silent failure that an user did not expect, so that can introduce a mistake. What's the reason behind this change?

Reply:
Thank you for pointing out this. Most of Kaggle data doesn't have any additional file describing category information separately, and hence I should add labels when reading annotation file line by line (each line contains individual dataset item). I agree with the existence of salient failure. Let me add a warning message at here.

Reply:
Although we need to expand the label category line by line during reading an annotation file, is it possible to keep the previous behavior? For example,

```python
label_category: LabelCategories

for line in file.read_lines():
     label_name: str = get_label(line)
     _, label = label_category.find(label_name)
     if label is None:
          label_cateogry.add(label_name)
```
It is reluctant to me changing drastically from the very strict check (assert) to the loose check (warning).

Reply:
I modified this without change annotation at 0b13690c5f9a0ad5bcf55e3d6b3baf17ed84c7eb.
It may be the same approach with your idea.